src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -330,7 +330,7 @@ private String escapeParse(StringTokenizer st,
         String fullName;
         nameFragment = firstToken;
         // skip spaces
-        while (" ".equals(nameFragment) && st.hasMoreTokens()) {
+        while ((0 == nameFragment.trim().length()) && st.hasMoreTokens()) {
```

Comment:
In this case, if we look at the caller of the method, the variable will never be null since it's a token returned from StringTokenizer (instead of returning null, NoSuchElementException will be thrown if there are no more tokens in this tokenizer).

You are right, it's a good practice to always check for null.